airbyte/sources/base.py

```diff
@@ -656,6 +669,46 @@ def read(
             )
 
         try:
+            # If max_lookback_days is provided, calculate the start_date and inject it into config
+            modified_config = None
+            if max_lookback_days is not None:
+                # Calculate the start_date by subtracting max_lookback_days from utcnow
+                start_date_dt = datetime.now(tz=timezone.utc) - timedelta(days=max_lookback_days)
+                start_date = start_date_dt.isoformat()
+
+                # Create a copy of the config and inject the start_date
+                modified_config = self.get_config().copy()
+                # Check if the connector config spec includes start_date before injecting it
+                # This avoids breaking connectors that don't use start_date
+                try:
+                    spec = self.config_spec()
+                    if "start_date" in spec.get("properties", {}):
+                        modified_config["start_date"] = start_date
+                    else:
+                        print(
+                            f"Warning: Connector does not support start_date configuration. "
+                            f"max_lookback_days={max_lookback_days} will be ignored."
+                        )
```

Comment:
Let's instead have a hard failure if "start_date" is not a permitted config input.